src/sentry/api/serializers/models/alert_rule.py

```diff
@@ -25,14 +28,16 @@
 
 @register(AlertRule)
 class AlertRuleSerializer(Serializer):
-    def __init__(self, expand=None):
+    def __init__(self, expand: list[str] | None = None):
         self.expand = expand or []
 
-    def get_attrs(self, item_list, user, **kwargs):
+    def get_attrs(
+        self, item_list: Sequence[Any], user: User | RpcUser, **kwargs: Any
+    ) -> defaultdict[AlertRule, Any]:
         alert_rules = {item.id: item for item in item_list}
         prefetch_related_objects(item_list, "snuba_query__environment")
 
-        result = defaultdict(dict)
+        result: defaultdict[AlertRule, dict[Any, Any]] = defaultdict(dict)
```

Comment:
I think this inner `dict[Any, Any]` is `dict[int, dict[str, Any]]` -- this should also be present in the returnvalue of `get_attrs` I think